### PR TITLE
[SEMI-MODULAR] Wrestling Rings

### PR DESCRIPTION
**Pull request by linnpap**
## About The Pull Request
dreamseeker_P0hrsDZK1t
IF YOU SMEEEEELALALALALALLLLLLLLL... WHAT THE ROCK... IS... COOKIN'

## Why It's Good For The Game
Glass fight cages look like shit.

## Changelog
:cl:
add: Added wrestling rings
/:cl:

### CODE REVIEW

**Comment by linnpap** [comment]
> 
> 
> You currently have a negative Fix/Feature pull request delta of -8. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

SHUT THE FUCK UP

File: code/game/objects/items/stacks/rods.dm [hunk 1]
```diff
@@ -6,6 +6,9 @@ GLOBAL_LIST_INIT(rod_recipes, list ( \
 	new/datum/stack_recipe("railing", /obj/structure/railing, 3, time = 18, window_checks = TRUE), \
 	new/datum/stack_recipe("tank holder", /obj/structure/tank_holder, 2, time = 5, one_per_turf = TRUE, on_floor = FALSE), \
 	new/datum/stack_recipe("ladder", /obj/structure/ladder/crafted, 15, time = 150, one_per_turf = TRUE, on_floor = FALSE), \
+	//SKYRAT EDIT ADDITION BEGIN - WRESTLING RING
+	new/datum/stack_recipe("wrestling ropes", /obj/structure/railing/wrestling, 3, time = 18, window_checks = TRUE), \
+	//SKYRAT EDIT ADDITION END
```
Comment (ZephyrTFA) [inline code/game/objects/items/stacks/rods.dm]: Don't put comments inside extended line long lists, put it after or before.
It fucks up the compiler hard

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 2]
```diff
@@ -125,6 +125,9 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
 	new/datum/stack_recipe("iron door", /obj/structure/mineral_door/iron, 20, one_per_turf = TRUE, on_floor = TRUE), \
 	new/datum/stack_recipe("floodlight frame", /obj/structure/floodlight_frame, 5, one_per_turf = TRUE, on_floor = TRUE), \
 	new/datum/stack_recipe("voting box", /obj/structure/votebox, 15, time = 50), \
+	//SKYRAT EDIT ADDITION - WRESTLING RING
+	new/datum/stack_recipe("wrestling turnbuckle", /obj/structure/wrestling_corner, 3, time = 25, one_per_turf = TRUE, on_floor = TRUE), \
+	//SKYRAT EDIT ADDITION END
```
Comment (ZephyrTFA) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: See above

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	if(I.tool_behaviour == TOOL_WELDER && !user.combat_mode)
+		if(obj_integrity < max_integrity)
+			if(!I.tool_start_check(user, amount=0))
+				return
+
+			to_chat(user, span_notice("You begin repairing [src]..."))
+			if(I.use_tool(src, user, 40, volume=50))
+				obj_integrity = max_integrity
+				to_chat(user, span_notice("You repair [src]."))
+		else
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm:96]: This else isn't needed

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+			to_chat(user, span_warning("[src] is already in good condition!"))
+		return
+
+/obj/structure/wrestling_corner/wirecutter_act(mob/living/user, obj/item/I)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm]: Use a better var than `I`

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+		return TRUE
+
+/obj/structure/wrestling_corner/deconstruct(disassembled)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		var/obj/item/stack/sheet/iron/iron_sheets = new /obj/item/stack/sheet/iron(drop_location(), 3)
+		transfer_fingerprints_to(iron_sheets)
+	return ..()
+
+///Implements behaviour that makes it possible to unanchor the railing.
+/obj/structure/wrestling_corner/wrench_act(mob/living/user, obj/item/I)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm]: See above

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+	. = ..()
+	if(flags_1&NODECONSTRUCT_1)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm]:
```suggestion
	if(flags_1 & NODECONSTRUCT_1)
```

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+	return . || mover.throwing || mover.movement_type & (FLYING | FLOATING)
+
+/obj/structure/wrestling_corner/proc/can_be_rotated(mob/user,rotation_type)
+	if(anchored)
+		to_chat(user, span_warning("[src] cannot be rotated while it is fastened to the floor!"))
+		return FALSE
+	return TRUE
+
+/obj/structure/wrestling_corner/proc/check_anchored(checked_anchored)
+	if(anchored == checked_anchored)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm]:
```suggestion
	return anchored == checked_anchored
```

File: modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+		return TRUE
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/wrestlingring/code/game/objects/structures/wrestlingring.dm]:
```suggestion
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/rods.dm (hunk 1 above)
Comment (linnpap) [reply code/game/objects/items/stacks/rods.dm]: i mean, there are countless other comments just like that in the list for recipes

**Comment by Gandalf2k15** [comment]
Fix suggested

**Pull request merged**